server/src/collection.rs

```diff
@@ -157,6 +157,29 @@ impl Collection {
         self.alias.lock().await.clone()
     }
 
+    pub async fn set_locked(&self, locked: bool) -> Result<(), ServiceError> {
+        let items = self.items.lock().await;
+
+        for item in items.iter() {
+            if !item.is_locked().await {
+                item.set_locked(true).await?;
+            }
+            let manager = self.manager.lock().await;
+            let signal_emitter = manager.signal_emitter(self.path.clone())?;
+            Self::item_changed(&signal_emitter, item.path()).await?;
+        }
+
+        if !self.is_locked().await {
```

Comment:
What is the condition for? So you cannot call set_locked if it is not locked?

Reply:
in other words, if there is some obvious reasoning, it is not that obvious and should be documented but I think it is just wrong.

Reply:
https://github.com/bilelmoussaoui/oo7/pull/150#discussion_r1846712451